src/mapgen_v6.cpp

```diff
@@ -420,13 +420,13 @@ BiomeV6Type MapgenV6::getBiome(int index, v2s16 p)
 	if (spflags & MGV6_SNOWBIOMES) {
 		float blend = (spflags & MGV6_BIOMEBLEND) ? noise2d(p.X, p.Y, seed) / 40 : 0;
 
-		if (d > FREQ_HOT + blend) {
-			if (h > FREQ_JUNGLE + blend)
+		if (d > (float)FREQ_HOT + blend) {
```

Comment:
Why are all these casts necessary? Apart from AVERAGE_MUD_AMOUNT, DESERT_STONE_BASE and DESERT_STONE_BASE they are already of type double. In addition to the casts being essentially useless, a much nicer solution is to simply define them as the correct type in the first place... E.g. #define FREQ_HOT 0.4**f**